src/graph_notebook/notebooks/04-Machine-Learning/Neptune-ML-SPARQL/neptune_ml_sparql_utils.py

```diff
@@ -0,0 +1,701 @@
+    os.makedirs(download_location, exist_ok=True)
+    # download embeddings and mapping info
+
+    S3Downloader.download(os.path.join(training_job_s3_output, "predictions/"),
+                          os.path.join(download_location, "predictions/"))
+
+    preds = np.load(os.path.join(download_location, "predictions", "result.npz"))['infer_scores']
+
+    if class_preds:
+        return preds.argmax(axis=1)
+
+    return preds
+
+
+def get_performance_metrics(training_job_name: str, download_location: str = './model-artifacts'):
+    training_job_s3_output = get_modeltraining_job_output_location(training_job_name)
+    if not training_job_s3_output:
+        return
+
+    download_location = os.path.join(download_location, training_job_name)
+    os.makedirs(download_location, exist_ok=True)
+    # download embeddings and mapping info
+
+    S3Downloader.download(os.path.join(training_job_s3_output, "eval_metrics_info.json"),
+                          download_location)
+
+    with open(os.path.join(download_location, "eval_metrics_info.json")) as f:
+        metrics = json.load(f)
+
+    return metrics
+
+
+class MovieLensProcessor:
+    raw_directory = fr'{HOME_DIRECTORY}/data/raw'
+    formatted_directory = fr'{HOME_DIRECTORY}/data/formatted'
+
+    ns_resource = Namespace("http://aws.amazon.com/neptune/resource#")
+    ns_ontology = Namespace("http://aws.amazon.com/neptune/ontology/")
+
+    def __download_and_unzip(self):
+        if not os.path.exists(f'{HOME_DIRECTORY}/data'):
+            os.makedirs(f'{HOME_DIRECTORY}/data')
+        if not os.path.exists(f'{HOME_DIRECTORY}/data/raw'):
+            os.makedirs(f'{HOME_DIRECTORY}/data/raw')
+        if not os.path.exists(f'{HOME_DIRECTORY}/data/formatted'):
+            os.makedirs(f'{HOME_DIRECTORY}/data/formatted')
+        # Download the MovieLens dataset
+        url = 'http://files.grouplens.org/datasets/movielens/ml-100k.zip'
+        r = requests.get(url, allow_redirects=True)
+        open(os.path.join(self.raw_directory, 'ml-100k.zip'), 'wb').write(r.content)
+
+        with zipfile.ZipFile(os.path.join(self.raw_directory, 'ml-100k.zip'), 'r') as zip_ref:
+            zip_ref.extractall(self.raw_directory)
+
+    def __upload_to_s3(self, bucketname: str):
+        path = urlparse(bucketname, allow_fragments=False)
+        bucket = path.netloc
+        file_path = path.path.lstrip('/').rstrip('/')
+
+        s3_client = boto3.client('s3')
+        for root, dirs, files in os.walk(self.formatted_directory):
+            for file in files:
+                s3_client.upload_file(os.path.join(
+                    self.formatted_directory, file), bucket, f'{file_path}/{file}')
+
+    def prepare_movielens_data_rdf(self, s3_bucket: str, clear_staging_area: bool):
+        self.formatted_directory = f'{self.formatted_directory}/rdf'
+        if clear_staging_area:
+            print('clearing staging area by default, use "clear_staging_area=False" to retain')
```

Comment:
nit - We should update this to use proper sentence capitalization since it will be seen by a user